src/test/container-features/e2e.test.ts

```diff
@@ -68,6 +72,16 @@ describe('Dev Container Features E2E (remote)', function () {
                 assert.equal(response.outcome, 'success');
                 assert.match(res.stderr, /CONTAINER ID/);
             });
+
+            it('should read configuration with features', async () => {
+                const res = await shellExec(`${cli} read-configuration --workspace-folder ${testFolder} --include-features-configuration  --skip-feature-auto-mapping`);
+                const response = JSON.parse(res.stdout);
+                console.log(res.stderr);
+                assert.equal(response?.featuresConfiguration?.featureSets[0]?.features[0]?.id, 'docker-in-docker');
+                assert.equal(response?.featuresConfiguration?.featureSets[0]?.features[0]?.customizations?.vscode?.extensions[0], 'ms-azuretools.vscode-docker');
+                assert.equal(response?.featuresConfiguration?.featureSets[2]?.features[0]?.id, 'node');
```

Comment:
Note that i do not think the order of this is stable, and might cause this test to incorrectly fail in the future

Reply:
Perhaps `response?.featuresConfiguration?.find(f => ...)` for each feature, and also adding an assert that the total size of `featureSets` is 3?

Reply:
Modified 👍